src/pybind/mgr/nfs/module.py

```diff
@@ -0,0 +1,169 @@
+            'desc': "Delete a cephfs export",
+            'perm': 'rw'
+        },
+        {
+            'cmd': 'nfs export ls '
+                   'name=clusterid,type=CephString '
+                   'name=detailed,type=CephBool,req=false ',
+            'desc': "List exports of a NFS cluster",
+            'perm': 'r'
+        },
+        {
+            'cmd': 'nfs export get '
+                   'name=clusterid,type=CephString '
+                   'name=binding,type=CephString ',
+            'desc': "Fetch a export of a NFS cluster given the pseudo path/binding",
+            'perm': 'r'
+        },
+        {
+            'cmd': 'nfs export update ',
+            'desc': "Update an export of a NFS cluster by `-i <json_file>`",
+            'perm': 'rw'
+        },
+        {
+            'cmd': 'nfs cluster create '
+                   'name=type,type=CephString '
+                   f'name=clusterid,type=CephString,goodchars={goodchars} '
+                   'name=placement,type=CephString,req=false ',
+            'desc': "Create an NFS Cluster",
+            'perm': 'rw'
+        },
+        {
+            'cmd': 'nfs cluster update '
+                   'name=clusterid,type=CephString '
+                   'name=placement,type=CephString ',
+            'desc': "Updates an NFS Cluster",
+            'perm': 'rw'
+        },
+        {
+            'cmd': 'nfs cluster delete '
+                   'name=clusterid,type=CephString ',
+            'desc': "Deletes an NFS Cluster",
+            'perm': 'rw'
+        },
+        {
+            'cmd': 'nfs cluster ls ',
+            'desc': "List NFS Clusters",
+            'perm': 'r'
+        },
+        {
+            'cmd': 'nfs cluster info '
+                   'name=clusterid,type=CephString,req=false ',
+            'desc': "Displays NFS Cluster info",
+            'perm': 'r'
+        },
+        {
+            'cmd': 'nfs cluster config set '
+                   'name=clusterid,type=CephString ',
+            'desc': "Set NFS-Ganesha config by `-i <config_file>`",
+            'perm': 'rw'
+        },
+        {
+            'cmd': 'nfs cluster config reset '
+                   'name=clusterid,type=CephString ',
+            'desc': "Reset NFS-Ganesha Config to default",
+            'perm': 'rw'
+        },
+    ]
+
+    MODULE_OPTIONS = []
+
+    def __init__(self, *args, **kwargs):
+        self.inited = False
+        self.lock = threading.Lock()
+        super(Module, self).__init__(*args, **kwargs)
+        with self.lock:
+            self.fs_export = FSExport(self)
+            self.nfs = NFSCluster(self)
+            self.inited = True
+
+    def handle_command(self, inbuf, cmd):
+        handler_name = "_cmd_" + cmd['prefix'].replace(" ", "_")
+        try:
+            handler = getattr(self, handler_name)
+        except AttributeError:
+            return -errno.EINVAL, "", "Unknown command"
+
+        return handler(inbuf, cmd)
```

Comment:
How to restrict characters with `@CLICommand` ? Can `goodchars` be passed as argument ?

Reply:
Nope, but given that's for a UUID we can do the following (importing `UUID` from `uuid`):
```diff
- def nfs_export_create_cephfs(self, fsname: str, clusterid: str, binding: str,
+ def nfs_export_create_cephfs(self, fsname: str, clusterid: UUID, binding: str,
```

And in `src/pybind/ceph_argparse.py`, which already has a `CephUUID` type:
```diff
    @staticmethod                                                                                          
    def to_argdesc(tp, attrs, has_default=False):                                                    
        if has_default:  
            attrs['req'] = 'false'    
        CEPH_ARG_TYPES = {                                                                                 
            str: CephString,                                                                               
            int: CephInt,                                                                                  
            float: CephFloat,                                                                              
            bool: CephBool,                                                                                 
+            UUID: CephUUID
        }   
```

Reply:
```
/home/varsha/ceph/build/bin/ceph -c /home/varsha/ceph/build/ceph.conf nfs cluster create cephfs vstart 
Error EINVAL: Traceback (most recent call last):
  File "/home/varsha/ceph/src/pybind/mgr/mgr_module.py", line 1337, in _handle_command
    return CLICommand.COMMANDS[cmd['prefix']].call(self, cmd, inbuf)
  File "/home/varsha/ceph/src/pybind/mgr/mgr_module.py", line 385, in call
    kwargs = self._collect_args_by_argspec(cmd_dict)
  File "/home/varsha/ceph/src/pybind/mgr/mgr_module.py", line 378, in _collect_args_by_argspec
    kwargs[k] = CephArgtype.cast_to(tp, v)
  File "/home/varsha/ceph/src/pybind/ceph_argparse.py", line 246, in cast_to
    return CephArgtype._cast_to_compound_type(tp, v)
  File "/home/varsha/ceph/src/pybind/ceph_argparse.py", line 231, in _cast_to_compound_type
    raise ValueError(f"unknown type '{tp}': '{v}'  {orig_type} {type_args}")
ValueError: unknown type '<class 'uuid.UUID'>': 'None'
```
It will require more changes to support UUID. I will look into this in a different PR. For now using regex instead.